crates/wysiwyg/src/dom/parser/parse.rs

```diff
@@ -321,7 +321,7 @@ mod sys {
             );
 
             match creation_result {
-                Ok(node) => node,
+                Ok(node) => DomNode::Mention(node),
```

Comment:
nitpick: We could implement `From` trait so that this becomes
```suggestion
                Ok(node) => node.into(),
```

Reply:
Will postpone this to a later bit of work, added a TODO in [46f48cb](https://github.com/matrix-org/matrix-rich-text-editor/pull/719/commits/46f48cb0f9e7c353e673b4e1179333f54bd8342c)